src/WindowsStateTriggers/EqualsStateTrigger.cs

```diff
@@ -70,27 +70,35 @@ internal static bool AreValuesEqual(object value1, object value2, bool convertTy
 			{
 				return true;
 			}
-			if (value1 != null && value2 != null)
+			if (value1 != null && value2 != null && value1.GetType() != value2.GetType() && convertType)
```

Comment:
Now that was really stupid of me to put it there.... :D Thanks for pointing out, removing it in my next commit!